### PR TITLE
Add blog picker to Gutenberg editor

### PR DESCRIPTION
**Pull request by pinarol**
Fixes part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/182

This change allows changing target blog while creating a new post:

screen shot 2018-11-22 at 15 13 15

**Implementation Details**

Please read [parent PR's](https://github.com/wordpress-mobile/WordPress-iOS/pull/10509) implementation details first.

PostEditorUtil is refactored as a PublishablePostEditor extension so we don't have a PostEditorUtil anymore.

And all PublishablePostEditor extensions are extracted to their own files according to their purpose.

So with this change we start to use PublishablePostEditor extensions to make aztec and gutenberg share same functionality.

The reason of doing this is PostEditorUtil and newly added PublishablePostEditor extensions are sharing the same purpose, they are both holding the logic that aztec and gutenberg have in common. So it is inconsistent to some of it in class and some of it in extensions. That's why I converted PostEditorUtil into an extension too.

**To test:** 

**Prerequisites:**

- clean pods folder
- run "rake dependencies"
- run yarn start on your mobile-gutenberg path
- open XCode and Option+Click Play(a.k.a Run) button
- Select "Run" option on the left
- Select Arguments on right and then check gutenberg option
- Click Run

**1. Test "Save as draft" - Change Target Blog**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Verify blog picker is opened
- Select a different site than current site
- Tap (...) -> Save as Draft
- Verify that Site picker button is turned into a label and you can no longer open the picker from there
- Tap Close
- Switch to that site
- Verify that the draft is added there

**2. Test "Save as draft" - Cancel Picker**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Tap cancel
- Verify that picker is closed

**3. Test "Publish" - Change Target Blog**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Select a different site than current site
- Tap publish
- Verify that post is closed
- Switch to that site
- Verify that the post is published there

**Aztec test**

- Deselect the gutenberg option in program arguments in Xcode and repeat the above 3 tests

### CODE REVIEW

**Comment by etoledom** [comment]
> PostEditorUtil is refactored as a PublishablePostEditor extension so we don't have a PostEditorUtil anymore.

I really like that change!

I have one observation about the name `PublishablePostEditor`. It seems that it now have much more than only publish related stuff. 

Do you think it would make sense to merge it with `PostEditor` ?
As the description of that protocol says:
> Common interface to all editors

**Comment by pinarol** [comment]
> Do you think it would make sense to merge it with PostEditor ?

Yes definitely, I was also planning to talk to you about that. Adding new content to PostEditor was actually what I did in the first place but then I saw that PostEditor was still being used else where so I didn't want to affect the purpose it was created then. And yes PublishablePostEditor is not just a "publishable" editor anymore, I agree that it is time to merge them 👍

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift [hunk 1]
```diff
@@ -399,32 +377,30 @@ extension PostEditorUtil {
...
-    }
+struct PostEditorDebouncerConstants {
+    static let autoSavingDelay = Double(0.5)
+}
 
-    struct Analytics {
-        static let headerStyleValues = ["none", "h1", "h2", "h3", "h4", "h5", "h6"]
-    }
+private struct Analytics {
+    static let headerStyleValues = ["none", "h1", "h2", "h3", "h4", "h5", "h6"]
```
Comment (etoledom) [inline WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: It looks like this is not being used.
If that's the case, can we remove it?

Comment (pinarol) [reply WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: right! removed 👍

**Pull request merged**